AMM/contracts/exchange/src/main.sw

```diff
@@ -0,0 +1,452 @@
+contract;
+
+dep errors;
+dep events;
+dep utils;
+
+use errors::{InitError, InputError, TransactionError};
+use events::{
+    AddLiquidityEvent,
+    DefineAssetPairEvent,
+    DepositEvent,
+    RemoveLiquidityEvent,
+    SwapEvent,
+    WithdrawEvent,
+};
+use libraries::{
+    data_structures::{
+        PoolInfo,
+        PreviewAddLiquidityInfo,
+        PreviewSwapInfo,
+        RemoveLiquidityInfo,
+    },
+    Exchange,
+};
+use std::{
+    block::height,
+    chain::auth::msg_sender,
+    context::{
+        call_frames::{
+            contract_id,
+            msg_asset_id,
+        },
+        msg_amount,
+    },
+    logging::log,
+    math::*,
+    storage::StorageMap,
```

Comment:
I know this is in the standard library however I have not checked which version it has been added. Can you double check the book release for the version of forc that we use to see if this needs to be specified or removed?

Reply:
It's in `forc 0.29.0` so will remove that while bumping the repo.